### PR TITLE
Add missing tests for graph.py

### PR DESCRIPTION
**Pull request by meatballs**

### CODE REVIEW

File: axelrod/tests/unit/test_graph.py [hunk 1]
```diff
@@ -86,6 +86,87 @@ def test_init(self):
...
 
+    def test_out_dict(self):
+        # Undirected graph with vertices and unweighted edges
+        g = graph.Graph(edges=[[1, 2], [2, 3]])
+        expected_out_mapping = {
+            1: {2: None},
+            2: {1: None, 3: None},
+            3: {2: None}
+        }
+        for key in expected_out_mapping:
```
Comment (marcharper) [inline axelrod/tests/unit/test_graph.py]: `assertDictEqual` is better since this loop just tests that one dict is a subset of the other.

Comment (meatballs) [inline axelrod/tests/unit/test_graph.py]: Yep. I forgot all about that assertion. Will do...

Comment (drvinceknight) [inline axelrod/tests/unit/test_graph.py]: Using `assertEqual` work just fine too for comparing dictionaries.

Comment (drvinceknight) [inline axelrod/tests/unit/test_graph.py]: From https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertDictEqual

> Test that two dictionaries are equal. If not, an error message is constructed that shows the differences in the dictionaries. This method will be used by default to compare dictionaries in calls to assertEqual().

**Comment by drvinceknight** [comment]
FYI `def out_dict` and `def in_dict` are still not covered (just mentioning it in case you wanted to grab them in this PR too: it can also wait for another one). :)

File: axelrod/tests/unit/test_graph.py (hunk 1 above)
Comment (meatballs) [inline axelrod/tests/unit/test_graph.py]: Cut and paste job from the previous code - but that had to cope with comparing a dict with a defaultdict. Not necessary here!

**Comment by meatballs** [comment]
> FYI def out_dict and def in_dict are still not covered (just mentioning it in case you wanted to grab them in this PR too: it can also wait for another one). :)

Whoops. That's exactly what these two tests were supposed to be for but I've got the assertion wrong. Hold on.....

**Comment by meatballs** [comment]
It wasn't the loop that shouldn't have been there - it was the pasted content inside the loop that was wrong!

**Comment by drvinceknight** [comment]
This also looks good to me.

**Pull request merged**